### PR TITLE
Circuit cutting: add interface integration tests

### PR DESCRIPTION
**Pull request by trbromley**
**Context:**

Adds tests for the `cut_circuit` transform that ensures differentiability with all of the interfaces.

### CODE REVIEW

**Comment by trbromley** [comment]
[sc-14724]

**Review by anthayes92 — APPROVED** [review]
Thanks for the interface tests! gradients are looking promising :grin:

File: tests/transforms/test_qcut.py [hunk 1]
```diff
@@ -1837,6 +1838,110 @@ def circuit(x):
...
+            qml.RX(x, wires=0)
+            qml.RY(0.543, wires=1)
+            qml.WireCut(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.RZ(0.240, wires=0)
+            qml.RZ(0.133, wires=1)
+            return qml.expval(qml.PauliZ(wires=[0]))
+
+        x = torch.tensor(0.531, requires_grad=True)
+        cut_circuit = qcut.cut_circuit(circuit, use_opt_einsum=use_opt_einsum)
```
[the remark this answers is not in the document]
Comment (josh146) [reply tests/transforms/test_qcut.py:1861]: (flyby comment: why not make it top-level, `@qml.cut_circuit`?)

File: tests/transforms/test_qcut.py [hunk 2]
```diff
@@ -1837,6 +1838,110 @@ def circuit(x):
...
+
+        res = cut_circuit(x)
+        res_expected = circuit(x)
+        assert np.isclose(res.detach().numpy(), res_expected.detach().numpy())
```
[the remark this answers is not in the document]
Comment (trbromley) [reply tests/transforms/test_qcut.py:1865]: If you do `assert np.isclose(res, res_expected)` then PyTorch gives the error:
```
RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.
```
Could possibly also use [torch.isclose](https://pytorch.org/docs/stable/generated/torch.isclose.html).

File: tests/transforms/test_qcut.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (trbromley) [reply tests/transforms/test_qcut.py:1861]: > is qcut.cut_circuit preferred over qml.transforms.cut_circuit?

For sure in user-facing documentation we don't want to go with `qcut` and should instead go through `qml`.

> (flyby comment: why not make it top-level, @qml.cut_circuit?)

Yes good idea, could do! What is the criteria for `qml.func()` vs `qml.transforms.func()`? It looks like it's grown a bit organically so far with both choices used.

**Review by trbromley — COMMENTED** [review]
Thanks @anthayes92 @josh146!

File: tests/transforms/test_qcut.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (josh146) [reply tests/transforms/test_qcut.py:1861]: > What is the criteria for qml.func() vs qml.transforms.func()? It looks like it's grown a bit organically so far with both choices used.

There is no criteria at the moment, it is done depending on UI considerations, and 'how important' a particular transform is!

**Pull request merged**